torch/_ops.py

```diff
@@ -470,6 +470,18 @@ def wrapper():
 
         return wrapper()
 
+    # NOTE [HigherOrderOp Schema]
+    # Each invocation of a HigherOrderOperator should have its own schema because
+    # the subgraphs and the arguments can be different even for the same hop.
+    # So each hop should implement its own gen schema method, which should
+    # take the same input as the __call__ method and returns a FunctionSchema.
+    # The schema can be used e.g. in Functionalization key to check if the hop
+    # mutates its inputs or have output aliasing input.
+    def gen_schema(self, *args, **kwargs):
+        raise NotImplementedError(
+            f"HigherOrderOp {self._name} does not implement a gen_schema."
```

Comment:
nit: spell out Operator, I know we don't in a lot of places but it's nicer to read